### PR TITLE
enable extension to expose static content

### PR DESCRIPTION
**Pull request by colombod**
kernel extensions can provide content if they implement the ```IStaticContentSource``` interface

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive/Events/KernelExtensionLoaded.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            ExtensionType = extensionType;
+            ContentSourceName = contentSourceName;
+            IsStaticContentSource = isStaticContentSource;
+        }
+        public KernelExtensionLoaded(IKernelExtension kernelExtension, KernelCommand command = null) : base(command)
+        {
+            KernelExtension = kernelExtension;
+            ExtensionType = kernelExtension.GetType().Name;
+
+            if (kernelExtension is IStaticContentSource contentSource)
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive/Events/KernelExtensionLoaded.cs:30]: This seems like extra bookkeeping.

File: src/dotnet-interactive.Tests/FileProviderTests.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+using FluentAssertions;
+
+using Microsoft.DotNet.Interactive.App.Http;
+using Microsoft.DotNet.Interactive.Tests;
+using Microsoft.DotNet.Interactive.Tests.Utility;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.Interactive.App.Tests
```
Comment (jonsequitur) [inline src/dotnet-interactive.Tests/FileProviderTests.cs:17]: To the extent you're testing interfaces directly and not over HTTP, it looks like these belong in `Microsoft.DotNet.Interactive.Tests`.

File: src/Microsoft.DotNet.Interactive/Events/KernelExtensionLoaded.cs (hunk 1 above)
Comment (colombod) [reply src/Microsoft.DotNet.Interactive/Events/KernelExtensionLoaded.cs]: removed

File: src/dotnet-interactive.Tests/FileProviderTests.cs (hunk 2 above)
Comment (colombod) [reply src/dotnet-interactive.Tests/FileProviderTests.cs:17]: that could bring too many asp.net dependencies

**Pull request merged**